### DISCUSSION TITLE
[Feature Request] Integrate Fingerprint Grow as an option

### DISCUSSION THREAD

**Discussion by mrbluebrett** [technyon/nuki_hub#559] [Ideas]
Would it be possible to integrate a Fiberprint Grow (like the R05) in the hub as an option?
I have this cheap fingerprint reader connected to an Esp32 with esphome nuki component. The fingerprint reader works very well, but the nuki component for esphome fails a lot and its kind of a nightmare. And this hub works very well so i think this can be a nice add-on.
It needs to connect via UART ON RX and TX gpios, and an optional sensing pin. Fingerprints are stored in the R503 itself, and it returns when a fingerprint matches and the id of the match, and with that maybe configure if we want to unlatch or unlock. Also we will need a new option in the UI to enroll new fingerprints or delete them.
The libraries for fingergrow are available for public in cpp.
I can provide help in the development and testing.
I hope this can be done!
Nice work and keep working on this project!

**Comment by iranl** [comment]
Functionality like this has been requested before. We don't intend to include functionality in NukiHub to work with specific 3rd party addons because of maintainability, stability, binary size, etc.

You can use another esp32 or smart home system to manage the output of your fingerprint sensor and if the logic there dictates performing an action on the lock use MQTT or GPIO with NukiHub.

**Reply by mrbluebrett** [reply]
The problem with using another esp and mqtt is it relays on stable wifi connection. No wifi, no unlatching.
And with gpio unlatching in nukihub, I tried time ago and has lots of false positives (even with a condenser) that opened my door when it should not.
But thanks anyway.

**Reply by iranl** [reply]
GPIO in NukiHub has had recent improvements to deal with false positives.

Networking can be done over ethernet if Wifi is troublesome.

**Comment by thebadboyltz** [comment]
I am in the same situation, the simplest is to put a relay on the esp32 of the Fiberprint Grow, this relay activates a gpio of the NuKiHub card. If the 2 esp32 use the same gnd, there is not even a need for a relay, the gpio of one card can activate the gpio of the other

**Reply by mrbluebrett** [reply]
I did this before, with both esps sharing gnd, but got lots of false positives. I will try again when I can to see if this has improved. But I need to be very sure there are no false positives opening my door…

**Reply by technyon** [reply]
@mrbluebrett The code has been improved, I got some advice from an electronics guy how to do the implementation. You have to hold the input signal now for a bit (about 300ms) to trigger the lock action, it will not be immediate as before (although for the user 300ms will be hardly noticeable). This delay also avoids the false positives.

Just one note: From a safety perspective this isn't great. If the fingerprint reader stores the fingerprint and decides wether or not to open the door, an intruder could just connect those wires from the outside as well.

**Reply by mrbluebrett** [reply]
Ok, maybe I will give it a try when I have time to test that.

And about safety, the fingerprint has a password (it can be changed whenever I want) so even if someone manages to open it and connect the wires it can't do anything without the password. And it's the esp32 (located inside my house) who really opens the door based on receiving an OK fingerprint match from the fingerprint.

**Reply by technyon** [reply]
In that case it should be fine.

**Reply by mrbluebrett** [reply]
I’m testing and at this moment i don’t have false positives. Thank you!

**Comment by thebadboyltz** [comment]
I haven't had time to make mine yet, but in this case use a 5v relay in this case you are sure to never have a positive fault

**Reply by mrbluebrett** [reply]
Yes, but 2 esps and a relay are much bulkier. I dont think i can fit all in the electric box i need to mount them.
